docs/sets.rst

```diff
@@ -0,0 +1,307 @@
+along; there is nothing to see here.
+
+
+Union and intersection
+######################
+
+Now that we know how to create and modity a set, let's learn how to evaluate the
+union and intersection of two sets. If you have two sets **A** and **B**, then
+the union is the combination of all elements from the two sets and in math is
+denoted as :math:`A \bigcup B`.
+The intersection is the sets of elements inside both A and B, and is denoted as
+:math:`A \bigcap B`.
+
+
+To see these in action, let's look at the integers from 1 through 10.
+
+.. code-block:: python
+
+    odds = set([1, 3, 5, 7, 9])
+    evens = set([2, 4, 6, 8, 10])
+    primes = set([2, 3, 5, 7])
+    composites = set([4, 6, 8, 9, 10])
+
+
+The union of the odd and even integers are all numbers from 1 to 10. You get the
+same answer if you reverse everything.
+
+.. code-block:: console
+
+    >>> odds.union(evens)
+    {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
+    >>> evens.union(odds)
+    {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
+
+    >>> odds
+    {1, 3, 5, 7, 9}
+    >>> evens
+    {2, 4, 6, 8, 10}
+
+
+Notice how the set of *odds* and the set of *evens* are unchanged. We can find
+the set of odd prime numbers by computing the intersection of the sets of
+**odds** and **primes**.
+
+.. code-block:: console
+
+    >>> odds.intersection(primes)
+    {3, 5, 7}
+    >>> primes.intersection(evens)
+    {2}
+
+
+And there is only one even prime number: **2**.
+
+Which integers are both *even* and *odd* ?
+
+.. code-block:: console
+
+    >>> evens.intersection(odds)
+    set()
+
+
+There are none. The intersection of these two sets is the empty set.
+
+.. code-block:: console
+
+    >>> primes.union(composites)
+    {2, 3, 4, 5, 6, 7, 8, 9, 10}
+
+
+The union of the prime numbers and composite numbers are the integers from **2**
+through **10**. Notice 1 is missing -- this is becausse 1 is neither prime nor
+composite.
+
+
+Membership
+##########
+
+
+Another common opreation is testing to see if one element is inside a set. To do
+this in Python use the **in** operator. Is 2 in the set of prime numbers ?
+
+
+.. code-block:: console
+
+    >>> 2 in primes
+    True
+
+
+Yes. This is a true statement.
+
+Is 6 an odd integer?
+
+.. code-block:: console
+
+    >>> 6 in odds
+    False
+
+
+No. This is a false statement. You can also test to see if an element is NOT in
+a set.
+
+.. code-block:: console
+
+    >>> 9 not in evens
+    True
+
+
+9 is *not* an even integer, so this is a true statement.
+
+There are many more methods and operations you can perform with sets. Take a
+moment to explore
+`these methods <https://www.w3schools.com/python/python_ref_set.asp>`_.
+You will not regret it.
+
+
+Sets are a built-in data type in Python. They come equipped with all the luxury
+features: unions, intersections, adding elements, removing elements, and much
+more. Everything you will ever need for your data hungry code... Provided your
+sets are *finite*.
```

Comment:
What do you mean by last sentence?

Reply:
In teorie multimile pot fi si infinite, gen: multimea numerelor prime, toate numerele impare, etc...